### PR TITLE
Add Artifactory docs

### PR DESCRIPTION
**Pull request by harrietgrace**
Adds the docs required for our Artifactory partnership. @toolmantim here's the draft so far - are you still keen on NPM for the example upload/download?

To do:

- [x] Pick a screenshot for the Uploading/downloading section
- [x] Add screenshots
- [x] Make the CLI examples more realistic
- [x] Add Artifactory notes to artifact cli page
- [x] Add Artifactory vars to the Agent --help docs example
- [x] PR Agent --help update
- [x] Add to sidebar in docs repo
- [ ] Add to sidebar in bk/bk repo

### CODE REVIEW

**Comment by toolmantim** [comment]
## Configuring package management

    To configure your builds to use [Artifactory's package management]( https://www.jfrog.com/confluence/display/RTF/Package+Management) for managing and installing dependencies, something something.

    For example, to configure all `npm` commands on your agent to use an [Artifactory NPM registry](https://www.jfrog.com/confluence/display/RTF/npm+Registry?src=sidebar), you can create an agent hook:

    ```bash
    #!/bin/bash
    set -e
    export ARTIFACTORY_NPM_REGISTRY="http://my-artifactory-server.internal:8081/artifactory/api/npm/npm-repo/"
    npm config set registry "$ARTIFACTORY_NPM_REGISTRY"
    ```

    To configure a Docker based build step to use the same NPM registry, you should pass the `ARTIFACTORY_NPM_REGISTRY` environment variable into the container, for example:

    ```yml
    steps:
      - plugins:
          docker#v1.2.3:
            image: node
            environment:
              - NPM_CONFIG_REGISTRY: "$ARTIFACTORY_NPM_REGISTRY"
    ```

    You can follow a similar pattern for Ruby, Docker, etc. (links to Artifactory)

**Comment by toolmantim** [comment]
Once you've configured the Artifactory buildkite agent environment variables, `artifact_paths` will automatically use artifactory. For example, the following will build a golang binary and upload it to artifactory, be visible in the Buildkite web, and can be downloaded in future steps using `buildkite-agent artifact download`.

    ```yml
    steps:
      - label: ":golang: :package:"
        command: "go build -v -o myapp-darwin-amd64"
        artifact_paths: "myapp-darwin-amd64"
        plugins:
          - docker#v1.2.3:
              image: "golang:1.8"
    ```

    You can also use [buildkite-agent artifact](...) commands directly, for example `buildkite-agent artifact upload my-file` and `buildkite-agent artifact download my-file`.

File: pages/integrations/artifactory.md.erb [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+
+
+You can also use [buildkite-agent artifact](https://buildkite.com/docs/agent/v3/cli-artifact) commands directly, for example `buildkite-agent artifact upload my-file` and `buildkite-agent artifact download my-file`.
+
+<%= image "buildkite-artifact-step.png", width: 2320/2, height: 822/2, alt: "Screenshot of a Buildkite command step's output logging an artifact upload to Artifactory" %>
+
+<%= image "artifactory-go-local-repository.png", width: 1486/2, height: 710/2, alt: "Screenshot of an artifact in the go-local repository in Artifactory" %>
+
+All artifacts in Buildkite link to their location in Artifactory. Click through to view them in Artifactory.
```
Comment (harrietgrace) [inline pages/integrations/artifactory.md.erb]: @toolmantim is this true? The go binary I uploaded with the example code doesn't link through to Artifactory - if you click it, it just gets downloaded to your computer 🤔

Comment (toolmantim) [reply pages/integrations/artifactory.md.erb]: Hrmm, is there an `BUILDKITE_ARTIFACTORY_ACCESS_URL` setting similar to `BUILDKITE_S3_ACCESS_URL`?

Comment (harrietgrace) [reply pages/integrations/artifactory.md.erb]: Not as far as I'm aware, I couldn't see one defined in the code alongside the other `BUILDKITE_ARTIFACTORY_*` variables. Should there be one?

Comment (toolmantim) [reply pages/integrations/artifactory.md.erb]: I think it's `BUILDKITE_S3_ACCESS_URL` that sets the URL endpoint if you want to be able to link to it directly from the Buildkite web UI?

**Comment by toolmantim** [comment]
I did a big review of it @harrietgrace, and ended up doing some edits because the fact there the env vars were all required in the end, and https://github.com/buildkite/agent/pull/1042 eventually landing on http://buildkite.localhost/docs/agent/v3/cli-artifact, it felt weird as lots of sections.

Let me know what you reckon! I noticed it's still missing a screenshot (but was hidden from output because it used angle brackets)

**Comment by harrietgrace** [comment]
We probably won't add in that extra screenshot. Unless you're keen to get that access_url var added into the agent for Artifactory? Otherwise you can't click through the link in the Artifacts tab, which was what we were going to show in that screenshot.

**Comment by toolmantim** [comment]
I thought `BUILDKITE_ARTIFACTORY_URL` makes the links work already? If not, bummer that's not supported! We can just remove that placeholder image.

**Comment by harrietgrace** [comment]
Nope 😬I tried it with my test pipeline and it doesn't link. I'll remove the screenshot placeholder, then this is probably ready for a merge?

**Comment by toolmantim** [comment]
Yep, I reckon it's ready to merge! ✨

**Pull request merged**